sunpy/map/maputils.py

```diff
@@ -1,7 +1,259 @@
+    map is not all off disk. If both these conditions are met, the
+    function returns `True`. Otherwise, the function returns `False`.
+
+    Parameters
+    ----------
+    smap : `~sunpy.map.GenericMap`
+        A map in helioprojective Cartesian coordinates.
+
+    Returns
+    -------
+    `~bool`
+        Returns `False` if any of the edge pixels are less than one solar
+        radius away from the center of the Sun.
+
+    Notes
+    -----
+    * The function assumes that the input map is rectangular.
+    * In the case of coronagraph images the disk itself need not be observed.
+
+    """
+    # Calculate all the edge pixels
+    edges = map_edges(smap)
+    edge_pixels = list(chain.from_iterable([edges["lhs"], edges["rhs"], edges["top"], edges["bottom"]]))
+    x = [p[0] for p in edge_pixels] * u.pix
+    y = [p[1] for p in edge_pixels] * u.pix
+
+    # Calculate the edge of the world
+    edge_of_world = smap.pixel_to_world(x, y)
+
+    # Calculate the distance of the edge of the world in solar radii
+    distance = u.R_sun * np.sqrt(edge_of_world.Tx ** 2 + edge_of_world.Ty ** 2) / smap.rsun_obs
+
+    # Test if all the edge pixels are more than one solar radius distant
+    # and that the whole map is not all off disk.
+    return np.all(distance > 1*u.R_sun) and ~is_all_off_disk(smap)
+
+
+@u.quantity_input
+def coordinate_is_on_disk(coordinate, scale: u.arcsecond):
+    """
+    Checks if the helioprojective Cartesian coordinate is on disk.
+
+    The check is performed by comparing the coordinate's distance
+    from the center of the Sun to the solar radius.
+
+    Parameters
+    ----------
+    coordinate : `~astropy.coordinates.SkyCoord`, `~sunpy.coordinates.frames.Helioprojective`
+        The input coordinate. The coordinate frame must be
+        `~sunpy.coordinates.Helioprojective`.
+
+    scale : `~astropy.units.Quantity`
+        The pixel scale size in units of arcseconds.
+
+    Returns
+    -------
+    `~bool`
+        Returns `True` if the coordinate is on disk, `False` otherwise.
+    """
+    # Calculate the radii of every pixel in helioprojective Cartesian
+    # co-ordinate distance units.
+    return u.R_sun * (np.sqrt(coordinate.Tx ** 2 + coordinate.Ty ** 2) / scale) < 1 * u.R_sun
+
+
+def is_all_off_disk(smap):
+    """
+    Checks if the entire `~sunpy.map.Map` is off the solar disk.
+
+    The check is performed by calculating the distance of every
+    pixel from the center of the Sun. If they are all off-disk,
+    then the function returns `True`. Otherwise, the function
+    returns `False`.
+
+    Parameters
+    ----------
+    smap : `~sunpy.map.GenericMap`
+        A map in helioprojective Cartesian coordinates.
+
+    Returns
+    -------
+    `~bool`
+        Returns `True` if all map pixels are strictly more than one solar radius
+        away from the center of the Sun.
+    """
+    return np.all(~coordinate_is_on_disk(all_coordinates_from_map(smap), smap.rsun_obs))
```

Comment:
This function would strangely return False for coronagraph images. Not sure that we can do anything about that...

Reply:
False is the correct answer.  Admittedly the Sun is behind an occulting disk, so the disk is not imaged. I think a previous version of this docstring did contain a comment about coronagraphs.  I can put that back in.

Reply:
I think it would make sense to add that proviso as a `note` so it doesn't surprise anyone.

Reply:
done